### DISCUSSION TITLE
Lost commit history (graph) when merging two accounts

### DISCUSSION THREAD

**Discussion by AlbinDaleby** [community/community#44095] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

I followed this guide https://scottmathson.com/blog/2019/02/13/combine-merge-github-user-accounts-keeping-git-commit-history/ to merge my work and private github account.

From what I understand any commit made is linked to the email address and not the github account itself.

I seem to have lost all commit history made by my work email when I connected it to my private account.
* I have been invited to my work organization again and I can see/browse all repositories and commit/push to them.
* I see new commits but as stated before the commit history was not automatically moved.

Might be because of a "stale-cache" according to this comment: https://stackoverflow.com/a/15290899
Any ideas what I can do to get all my historically made commits?

**Comment by bzkarim250** [comment]
When you merge two Github accounts, the commit history is not automatically moved. The commits are linked to the email address and not the Github account, so when you change the association of an email address from one account to another, the commit history will not be transferred.     One way to regain the commit history is to use the `git filter-branch` command to change the author and committer information of the commits to match the new account.

**Reply by AlbinDaleby** [reply]
Hello.

This is what I did:
1. Added a 2nd email to my work account
2. Removed my work email from work account
3. Added work to my private account
4. Got invited to my organisation and repos.

So if the commits are linked to the email. Why dont I see the work email commits on my main account, which now have my work email and my private email in the Email list?

Thanks.

**Reply by jairus-m** [reply]
Any luck with this? I am currently trying to do the same thing.

**Reply by nadavjulius** [reply]
same

**Reply by jairus-m** [reply]
> same

Hey, just create a ticket for this issue via GitHub support and they'll sort you out! They got back to me within several days of requesting a fix.
